src/content/docs/tutorials/rust/social_network.mdx

```diff
@@ -0,0 +1,222 @@
+```
+Which returns:
+```
+coolwolf752, 3
+stormfox762, 5
+darkdog878, 6
+brightninja683, 5
+stormcat597, 2
+silentguy245, 1
+fastgirl798, 4
+stormqueen831, 4
+epiccat105, 4
+brightking765, 1
+mysticcat651, 4
+smartdragon25, 3
+epicwolf202, 3
+stormninja678, 6
+coolking201, 3
+fastqueen400, 2
+silentninja637, 2
+mysticwolf198, 4
+cooldragon866, 1
+epicking81, 3
+```
+This is a lot more useful! we can now clearly see how many follower each user has.
+3. Lastly, we use `ORDER BY` and `LIMIT` to order the usernames by their amount of followers, and return only the first entry. This is the user we are looking for, and how many followers that user has:
+```Rust
+    let result = conn.query("MATCH (u1:User)-[f:FOLLOWS]->(u2:User) RETURN u2.username, count(u2) as follower_count ORDER BY follower_count DESC LIMIT 1")?;
+    for row in result {
+        let row_str = row.iter()
+            .map(|col| col.to_string())  // Convert each column to a string
+            .collect::<Vec<String>>()    // Collect into a Vec<String>
+            .join(", ");                 // Join elements with ", "
+    
+        println!("{}", row_str);
+    } 
+```
+Returning:
+```
+darkdog878, 6
+```
+
+### Q2: What is the shortest path between two users?
+Another question which we might be interested in answering is finding the shortest path between two users. For example, how many follows does it take to get from user `silentguy245` to `epicwolf202`?
+1. We may start off the query with a simple `MATCH` query followed by a `WHERE` query to specify the results we want:
+```Rust
+    conn.query("""
+        MATCH p = (u1:user)-[f:FOLLOWS]->(u2:User)
+        WHERE u1.username = 'silentguy245' AND u2.username = 'epicwolf202'
+        RETURN p;
+    """)?;
+```
+This will return us whether `silentguy245` follows `epicwolf202`, which they don't. 
+2. We can then build off this query using [recursive match](../../cypher/query-clauses/match.md) to find path instead:
+```Rust
+    conn.query("""
+        MATCH p = (u1:user)-[f:FOLLOWS* SHORTEST 1..4]->(u2:User)
+        WHERE u1.username = 'silentguy245' AND u2.username = 'epicwolf202'
+        RETURN p;
+    """)?;
+```
+This will return us a result in `RECURSIVE_REL` datatype, which is not as useful to interpret.
+3. We will then simplify the result using some [recursive relationship functions](../../cypher/expressions/recursive-rel-functions.md)
+```Rust
+    conn.query("""
+        MATCH p = (u1:user)-[f:FOLLOWS* SHORTEST 1..4]->(u2:User)
+        WHERE u1.username = 'silentguy245' AND u2.username = 'epicwolf202'
+        RETURN properties(nodes(p), 'username');
+    """)?;
+```
+
+### Q3: How many 3-hop paths exist between userA and userB that passes through userC?
+To answer this query, we should do something similar to Q1. We first match all 3-hop queries:
+```Rust
+    conn.query("""
+        MATCH (u1:User)-[f1:FOLLOWS]->(u2:User)-[f2:Follows]->(u3:User)-[f3:FOLLOWS]->(u4:User)
+        RETURN count(u4);
+    """)?;
+```
```

Comment:
We should always show the results of every query. Exactly as it would appear in the terminal. So that the user knows what to expect.